packages/exporter-trace-otlp-http/src/OTLPExporterBase.ts

```diff
@@ -102,6 +112,32 @@ export abstract class OTLPExporterBase<
     });
   }
 
+  private _configureTimeout(timeoutMillis: number | undefined): number {
+    return typeof timeoutMillis === 'number'
+      ? timeoutMillis < 0
```

Comment:
discussion:
I wonder what should be the semantic meaning of timeout `0`.
If the user sets the value to 0, it will always timeout before there is any chance for the export to report real status which doesn't make a lot of sense to me. 
Sometimes the value 0 for timeout is interpreted as "disable" or "no timeout at all".
Not sure what is the expected behavior here. I wish the spec stated it but looks like it does not.

Reply:
I wasn't sure what to do with timeout when it's set to `0` since the spec mentions warning the user only when it's negative. But I do feel like if the user sets `0` then we should use warn them about that timeout value and use the `DEFAULT_TIMEOUT` value (10000ms) instead.

Reply:
If we do decide to forward the timeout to the `http` package, then we can just fallback to it's implementation.
I couldn't find how timeout of 0 is handled, but I guess it's being forwarded to the os [where timeout 0 means "no timeout"](https://linux.die.net/man/3/setsockopt):

> SO_RCVTIMEO
> Sets the timeout value that specifies the maximum amount of time an input function waits until it completes. It accepts a timeval structure with the number of seconds and microseconds specifying the limit on how long to wait for an input operation to complete. If a receive operation has blocked for this much time without receiving additional data, it shall return with a partial count or errno set to [EAGAIN] or [EWOULDBLOCK] if no data is received. **The default for this option is zero, which indicates that a receive operation shall not time out**. This option takes a timeval structure. Note that not all implementations allow this option to be set.

Anyway, it's a minor detail, so feel free to implement whatever you think is right and resolve the conversation when you are settled :)

Reply:
I went with checking if timeout of 0 is set by the user and if it is, then just display the warning and use default timeout as specified in the environment.ts file.